### PR TITLE
Add TSLint security plugin report files and tests

### PR DESCRIPTION
**Pull request by MVrachev**
Part of the work outlined in the enhancement issue: https://github.com/vmware/precaution/issues/160

I added the report generation from the TSLint plugin output.

I added three unit tests.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

### CODE REVIEW

File: tslint/tslint_annotations.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { getAnnotationLevel } = require('../annotations_levels')
+
+/**
+ * Translates the tslint severity from "ERROR" | "WARNING" | "OFF"
+ * to the "HIGH", "MEDIUM" and "LOW"
+ * @param {String} severity issues severity given by tslint
+ */
+function translateSeverity (severity) {
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: Aren't we doing similar mappings like this in annotations_levels.js?

File: tslint/tslint_annotations.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  switch (severity) {
+    case 'ERROR': result = 'HIGH'; break
+    case 'WARNING': result = 'MEDIUM'; break
+    case 'OFF': result = 'LOW'
+  }
+  return result
+}
+
+/**
+ * Create bandit Annotations
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: bandit?

File: tslint/tslint_annotations.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {*} issue
+ * @param {String} directory working directory for the Gosec process
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: Gosec?

File: tslint/tslint_annotations.js (hunk 2 above)
Comment (MVrachev) [reply tslint/tslint_annotations.js]: Yes, a mistake.

File: tslint/tslint_annotations.js (hunk 3 above)
Comment (MVrachev) [reply tslint/tslint_annotations.js]: Same here.

File: tslint/tslint_annotations.js (hunk 1 above)
Comment (MVrachev) [reply tslint/tslint_annotations.js]: We don't have to.
The severity attribute in the output from Bandit and Gosec is "HIGH", "MEDIUM" or "LOW" like the ones used in the annotations_levels.js.
This is a specific situation in this linter.

But there is another annoying thing. There is no "confidence" in the output and I give "HIGH" confidence by default...

Comment (ericwb) [reply tslint/tslint_annotations.js]: So yes, `annotations_levels.js` looks at confidence whereas TSLint has none, but that's a very easy change to use a `default` in the switch statement for cases where there is no confidence.

Also, looks odd to map `OFF` to `LOW` since I assume `OFF` indicates you'll get no results.  Although I'm not clear what happens with `OFF` according to the documentation.

Comment (MVrachev) [reply tslint/tslint_annotations.js]: So we should ignore the issues with annotations off?

I will speak with the TSLint team about it.

Comment (MVrachev) [reply tslint/tslint_annotations.js]: When there is no confidence from a linters output what should be the default value? HIGH?

File: tslint/tslint_annotations.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+/**
+ * Create TSLint Annotations
+ * @param {*} issue
+ * @returns {Object} see https://developer.github.com/v3/checks/runs/#annotations-object-1
+ */
+function getAnnotation (issue) {
+  const path = issue.name
+  // I add 1 to startLine and endline because they are off by 1 as described here:
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: Who is I?  Try to avoid using first person in comments

File: tslint/tslint_annotations.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * Translates the tslint severity from "ERROR" | "WARNING" | "OFF"
+ * to the "HIGH", "MEDIUM" and "LOW"
+ * @param {String} severity issues severity given by tslint
+ */
+function translateSeverity (severity) {
+  let result = ''
+  switch (severity) {
+    case 'ERROR': result = 'HIGH'; break
+    case 'WARNING': result = 'MEDIUM'; break
+    case 'OFF': result = 'LOW'
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: It's still not clear to me what OFF severity means.

File: annotations_levels.js [hunk 6]
```diff
@@ -38,20 +38,23 @@ function getAnnotationLevel (severity, confidence) {
         case 'HIGH' : result = annotationsLevels.severityHIGHconfidenceHIGH; break
         case 'MEDIUM' : result = annotationsLevels.severityHIGHconfidenceMEDIUM; break
         case 'LOW' : result = annotationsLevels.severityHIGHconfidenceLOW; break
+        default: result = annotationsLevels.severityHIGHconfidenceHIGH
```
Comment (ericwb) [inline annotations_levels.js]: Since we're not using these changes for this PR, I would ask you put this in a separate PR.

File: tslint/tslint_annotations.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { getAnnotationLevel } = require('../annotations_levels')
+
+/**
+ * Translates the tslint severity from "ERROR" | "WARNING" | "OFF"
+ * to the "HIGH", "MEDIUM" and "LOW"
+ * @param {String} severity issues severity given by tslint
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: What is meant by "severity issues severity"

File: tslint/tslint_annotations.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * Create TSLint Annotations
+ * @param {*} issue
+ * @returns {Object} see https://developer.github.com/v3/checks/runs/#annotations-object-1
+ */
+function getAnnotation (issue) {
+  const path = issue.name
+  // I add 1 to startLine and endline because they are off by 1 as described here:
+  // https://github.com/palantir/tslint/issues/4528
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: I also generally don't add comments referring to bugs since that is what git log and git blame is for.

File: annotations_levels.js (hunk 6 above)
Comment (MVrachev) [reply annotations_levels.js]: I just realized that we are giving a default annotation level when we initialize the result value in the begging of the function. The default annotation level when there is no confidence provided is "warning" whatever the severity is.

Do we want to give other default annotation level based on the severity when there is no confidence provided?

Comment (MVrachev) [reply annotations_levels.js]: I will remove the default statements from this switch for now.

File: tslint/tslint_annotations.js (hunk 7 above)
Comment (MVrachev) [reply tslint/tslint_annotations.js]: I will paraphrase it.

File: tslint/tslint_annotations.js (hunk 5 above)
Comment (MVrachev) [reply tslint/tslint_annotations.js]: To we fair I am not sure after spending a few hours investigating. 
I opened an issue about it: https://github.com/palantir/tslint/issues/4634

File: tslint/tslint_annotations.js (hunk 4 above)
Comment (MVrachev) [reply tslint/tslint_annotations.js]: Yep, you are right. I will change it.

File: tslint/tslint_report.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { getAnnotation } = require('./tslint_annotations')
+const { countIssueLevels } = require('../annotations_levels')
+
+/**
+ * Process TSLint output (generate annotations, count issue levels)
+ * @param {*} results TSLint json output
+ */
```
Comment (ericwb) [inline tslint/tslint_report.js]: Missing doc on return

File: tslint/tslint_annotations.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { getAnnotationLevel } = require('../annotations_levels')
+
+/**
+ * Translates the tslint severity from "ERROR" | "WARNING" | "OFF"
+ * to the "HIGH", "MEDIUM" and "LOW"
+ * @param {String} severity the severity of the issue provided by  TSLint
+ */
```
Comment (ericwb) [inline tslint/tslint_annotations.js]: Missing doc about the return value
